tenseal/tensors/ckksvector.cpp

```diff
@@ -352,4 +376,32 @@ CKKSVector& CKKSVector::matmul_plain_inplace(
     return *this;
 }
 
+CKKSVector CKKSVector::replicate_first_slot(size_t n) {
+    CKKSVector new_vector = *this;
+    return new_vector.replicate_first_slot_inplace(n);
+}
+
+CKKSVector& CKKSVector::replicate_first_slot_inplace(size_t n) {
+    // mask
+    vector<double> mask = {1};
+    // TODO: remove this after resolving issue of transparent ciphertext
+    // which adds a 1 at the end
+    for (size_t i = 0; i < n - 1; i++) mask.push_back(0);
```

Comment:
Why not
vector<double> mask(n, 0);
mask[0] = 1;
?

Reply:
True!